src/Illuminate/Database/Concerns/BuildsQueries.php

```diff
@@ -147,6 +149,30 @@ public function first($columns = ['*'])
         return $this->take(1)->get($columns)->first();
     }
 
+    /**
+     * Execute the query and get the first result if it's the sole.
+     *
+     * @param  array|string  $columns
+     * @return \Illuminate\Database\Eloquent\Model|object|static|null
+     *
+     * @throws \Illuminate\Database\NoRecordsFoundException
+     * @throws \Illuminate\Database\MultipleRecordsFoundException
+     */
+    public function sole($columns = ['*'])
+    {
+        $result = $this->get($columns);
```

Comment:
Shouldn't we always force it to `take(2)` instead of relying on the user's criteria (which could potentially load many models in memory)?

Also, although the docblocks state the exceptions, shouldn't we adhere to the `*OrFail` convention?

Not trying to nitpick here, just thought it would make an interesting discussion. ;)

Reply:
Good call on `take()` added that. As for naming, I leave that to Taylor :)

Reply:
Isn't `*OrFail` needed only if there's a silent version of the method ? (As rails add a bang "!" version of methods only if there is a non bang version ?)

Reply:
Yes, although in this case the "silent" version of `sole` would be the same as the `first()`, so we wouldn't need its silent version, I think.

Reply:
Ah yes, didn't see it that way. And it doesn't hurt to be explicit anyway, especially since it's a convention that already exists in the framework.
Either way, I'm happy with the result!